templates/prometheus.go

```diff
@@ -45,25 +52,62 @@ var PrometheusTemplate = promv1.Prometheus{
 				Port:      intstr.FromString("web"),
 			}},
 		},
-		Resources: utils.GetResourceRequirements("prometheus"),
-		// prometheus tls -- will be uncommented
-		// Web: &promv1.WebSpec{
-		// 	TLSConfig: &promv1.WebTLSConfig{
-		// 		KeySecret: corev1.SecretKeySelector{
-		// 			Key: "tls.key",
-		// 		},
-		// 		Cert: promv1.SecretOrConfigMap{
-		// 			Secret: &corev1.SecretKeySelector{
-		// 				Key: "tls.crt",
-		// 			},
-		// 		},
-		// 		ClientAuthType: fmt.Sprint(tls.VerifyClientCertIfGiven),
-		// 		ClientCA: promv1.SecretOrConfigMap{
-		// 			ConfigMap: &corev1.ConfigMapKeySelector{
-		// 				Key: "service-ca.crt",
-		// 			},
-		// 		},
-		// 	},
-		// },
+		Resources:   utils.GetResourceRequirements("prometheus"),
+		ListenLocal: true,
+		Containers: []corev1.Container{{
+			Name: "kube-rbac-proxy",
+			Args: []string{
+				"--secure-listen-address=0.0.0.0:9339",
+				"--upstream=http://127.0.0.1:9090/",
+				"--logtostderr=true",
+				"--v=10",
+				"--tls-cert-file=/etc/tls-secret/tls.crt",
+				"--tls-private-key-file=/etc/tls-secret/tls.key",
+				"--client-ca-file=/etc/tls-cert/service-ca.crt",
+				"--config-file=/etc/kube-rbac-proxy/config-file.yaml",
+			},
+			Ports: []corev1.ContainerPort{{
+				Name:          "https",
+				ContainerPort: 9339,
+			}},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      ServingCertificateKeyVolumeName,
+					MountPath: "/etc/tls-secret",
+				},
+				{
+					Name:      ServiceCACertificateVolumeName,
+					MountPath: "/etc/tls-cert",
+				},
+				{
+					Name:      KubeRbacProxyConfigVolumeName,
+					MountPath: "/etc/kube-rbac-proxy",
+				},
+			},
+		}},
+		Volumes: []corev1.Volume{
+			{
+				Name: ServingCertificateKeyVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{},
+				},
+			},
+			{
+				Name: ServiceCACertificateVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{},
+					},
+				},
+			},
+			{
+				Name: KubeRbacProxyConfigVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{},
+					},
+				},
+			},
+		},
```

Comment:
This needs to come out of the box in our prom package, the operator should not concern itself with adding containers into the Prometheus bundle/deployment.

Reply:
After discussing it with @bindrad I agree we need to add the container in code.